GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -312,6 +314,17 @@ public FormBrowse([NotNull] GitUICommands commands, string filter, ObjectId sele
 
             return;
 
+            void WorkaroundToolbarLocationBug()
+            {
+                // Layout engine bug (?) which may change the order of toolbars
+                // if the 1st one becomes longer than the 2nd toolbar's Location.X
+                // the layout engine will be place the 2nd toolbar first
+                toolPanel.TopToolStripPanel.Controls.Clear();
+                toolPanel.TopToolStripPanel.Controls.Add(ToolStripScripts);
```

Comment:
If the order is left-to-right display order, should the order consistently be Filters, Sscripts, Main?

and I would prefer to remove/add button by button instead, splitting the toolbar is not adding much...

Reply:
> splitting the toolbar is not adding much

It is adding if a) toolbar can be moved, and b) toolbars can be shown/hidden. "a" obviously doesn't work at this stage, but "b" is working just fine.

Reply:
> should the order consistently be Filters, Sscripts, Main?

The order is: Main, Filters, Scripts.